crates/teloxide/src/backoff.rs

```diff
@@ -2,14 +2,30 @@ use std::time::Duration;
 
 pub type BackoffStrategy = Box<dyn Send + Fn(u32) -> Duration>;
 
+const THRESHOLD_ERROR_QUANTITY: u32 = 6;
+
 /// Calculates the backoff time in seconds for exponential strategy with base 2
 ///
-/// The maximum duration is limited to a little less than half an hour (1024
-/// secs), so the successive timings are(in secs): 1, 2, 4, .., 1024, 1024, ..
+/// The maximum duration is limited to a little more than a minute: 64s, so the
+/// successive timings are: 1s, 2s, 4s, .., 64, .., 64
 ///
 /// More at: <https://en.wikipedia.org/wiki/Exponential_backoff#Exponential_backoff_algorithm>
 pub fn exponential_backoff_strategy(error_count: u32) -> Duration {
-    // The error_count has to be limited so as not to cause overflow: 2^10 = 1024 ~
-    // a little less than half an hour
-    Duration::from_secs(1_u64 << error_count.min(10))
+    // 2^6 = 64s ~ a little more than a minute
+    Duration::from_secs(1_u64 << error_count.min(THRESHOLD_ERROR_QUANTITY))
+}
+
+#[cfg(test)]
+mod tests {
+    use rstest::rstest;
+
+    use super::*;
+
+    #[rstest]
+    #[case(1, Duration::from_secs(2))]
+    #[case(5, Duration::from_secs(32))]
+    #[case(42, Duration::from_secs(64))]
```

Comment:
That is nice ig, but I feel like we shouldn't bring a dependency for something that can be done with just a for loop over an array ...